### ISSUE TITLE
about create executor when i add a task

### ISSUE THREAD

**Issue by sapluk** [srikanth-lingala/zip4j#100]
My code:
```
ZipFile zipFile = new ZipFile("/path/to/my.zip");
zipFile.setRunInThread(true);
ProgressMonitor progressMonitor = zipFile.getProgressMonitor();
zipFile.addFile(new File("/path/to/myFile"));
// ...
```

When I add a task, the utils uses `Executors.newSingleThreadExecutor()` to create a thread pool to perform the task. I can't get the object reference and can't shutdown it.
When I apply it to a service, every time I use this feature, it always create a thread pool that will take up the server's resources.

**Comment by srikanth-lingala** [comment]
To be clear: Would you like to have a reference to the executor?

**Comment by sapluk** [comment]
i want to shutdown it

**Comment by sapluk** [comment]
![image](https://user-images.githubusercontent.com/37981448/68591265-44f4fd80-04cb-11ea-9a35-b9bdc264f04e.png)

**Comment by srikanth-lingala** [comment]
I will expose a reference to the executor from the ZipFile api. Will include it in the next release. There was a request for similar feature in #80 , but had to close it as there was no further feedback from the user.

**Comment by srikanth-lingala** [comment]
Feature added. You can now get access to `ExecutorService` with `ZipFile.getExecutorService()`. Please note that the executor service is only initialised if zip4j is run in thread mode, and after starting of a process (ex: adding files to zip, etc). In other words:

```
zipFile.setRunInThread(true);
zipFile.getExecutorService(); // --> is null at this moment
zipFile.addFile(new File("somefile.txt));
```

```
zipFile.setRunInThread(true);
zipFile.addFile(new File("somefile.txt));
zipFile.getExecutorService(); // --> returns the executor service used by zip4j
```

I will include this feature in the next release.

**Comment by srikanth-lingala** [comment]
Feature included in v2.4.0 released today
